tools/cfn2ts/lib/genspec.ts

```diff
@@ -184,32 +188,35 @@ export function validatorName(typeName: CodeName): CodeName {
  * - The type we will generate for the attribute, including its base class and docs.
  * - The property name we will use to refer to the attribute.
  */
-export function attributeDefinition(resourceName: CodeName, attributeName: string, docLink?: string): Attribute {
+export function attributeDefinition(resourceName: CodeName, attributeName: string, spec: schema.Attribute): Attribute {
     const descriptiveName = descriptiveAttributeName(resourceName, attributeName);  // "BucketArn"
     const propertyName = cloudFormationToScriptName(descriptiveName);            // "bucketArn"
 
-    // Not in a namespace, base the name on the descriptive name
-    const typeName = new CodeName(resourceName.packageName, '', descriptiveName); // "BucketArn"
-    const baseClass = attributeName.endsWith('Arn') ? ARN_NAME : TOKEN_NAME;
+    let attrType;
+    if ('PrimitiveType' in spec && spec.PrimitiveType === 'String') {
+        attrType = new ClassDeclaration(CodeName.forPrimitive('string'));
+    } else {
+        // Not in a namespace, base the name on the descriptive name
+        const typeName = new CodeName(resourceName.packageName, '', descriptiveName); // "BucketArn"
+        const baseClass = attributeName.endsWith('Arn') ? ARN_NAME : TOKEN_NAME;
+
+        attrType = new ClassDeclaration(typeName, baseClass, undefined);
+    }
 
     const constructorArguments = `this.getAtt('${attributeName}')`;
-
-    const attrType = new ClassDeclaration(typeName, baseClass, docLink);
     return new Attribute(propertyName, attrType, constructorArguments);
 }
 
 /**
  * Return an attribute definition name for the RefKind for this class
  */
 export function refAttributeDefinition(resourceName: CodeName, refKind: string): Attribute {
-    const refClassName = descriptiveAttributeName(resourceName, refKind);
-    const refClass = new CodeName(resourceName.packageName, '', refClassName);
-    const baseClass = refKind === schema.SpecialRefKind.Arn ? ARN_NAME : TOKEN_NAME;
+    const propertyName = codemaker.toCamelCase(descriptiveAttributeName(resourceName, refKind));
 
-    const constructorArguments = '{ Ref: this.logicalId }, `${this.logicalId}.Ref`';
+    const constructorArguments = 'this.ref';
 
-    const refType = new ClassDeclaration(refClass, baseClass);
-    return new Attribute('ref', refType, constructorArguments);
+    const refType = new ClassDeclaration(CodeName.forPrimitive('string'));
```

Comment:
Actually the name `ClassDeclaration` is now wrong again, because we're never going to declare that class anymore.

Reply:
We still have non-primitive attributes that we are emitting as strong types. Renamed to `AttributeTypeDeclaration` for now.